### PR TITLE
My subscriptions action modals

### PR DESCRIPTION
**Pull request by bgrgicak**
### Changes proposed in this Pull Request:

This PR adds subscription actions. Depending on the subscription state, the update button will trigger a different modal (subscribe, renew, or connect).
Action buttons will now load in the actions column for each subscription and depending on the subscription state they can be install, subscribe, connect, or renew.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. [Connect your site to WCCOM](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions)
2. Confirm that the install button works
3. Click on the renew button and confirm it opens checkout on WCCOM with your renewal added to the cart (you will need an expired subscription for this)
4. Manually install an outdated version of a product for which you don't have a subscription
5. Click on the update button and confirm it asks you to subscribe first
6. Click on the subscribe button and confirm it opens cart on WCCOM with your product added to the cart
7. After you complete checkout [go back to my subscriptions ](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions)
8. Click connect and confirm that the subscription is connected

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

Add my subscription action buttons and modals.

</details>

Screenshot 2023-10-24 at 11 14 49

### CODE REVIEW

**Review by kdevnel — APPROVED** [review]
Epic PR! All the interactions worked exactly as described and as expected. Super impressed with this.

I have a comment that is mostly focussed on design and so isn't stopping me approving this. It might be worth getting a copy of this PR for Poli to test so we can get early feedback - she is setup to run PRs on a local install. I also discovered you can run a specific branch using the advanced settings on Jurassic.ninja which is also handy.

Anyway, the feedback is on the buttons. I see the primary style is used for renewal related buttons (renew, subscribe) and the secondary style without a button outline for the other buttons. This is okay, but feels a little weird with the `isBusy` state. Something about it doesn't feel quite right, but I can't pinpoint what. Again, it's small but might be best being checked out by Poli.

**Comment by bgrgicak** [comment]
> Again, it's small but might be best being checked out by Poli.

Let's check with @poligilad-auto. I avoided making css changes in this PR and focused on the features, but we can address css changes in another PR.

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss [hunk 1]
```diff
@@ -143,3 +144,15 @@
 		margin-bottom: $grid-unit-30;
 	}
 }
+
+.woocommerce-marketplace__my-subscriptions__table__header--actions {
+	text-align: right;
+}
+.woocommerce-marketplace__my-subscriptions__actions {
```
Comment (andfinally) [inline plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss:152]:
```suggestion

.woocommerce-marketplace__my-subscriptions__actions {
```

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/subscribe-button.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { MARKETPLACE_CART_PATH } from '~/marketplace/components/constants';
+import { appendURLParams } from '~/marketplace/utils/functions';
```
Comment (andfinally) [inline plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/subscribe-button.tsx]: The `~` looks weird, does it not mean the current user's home directory in this context? I see we do this also in `actions/connect-button.tsx`. It does seem to work.

**Review by andfinally — APPROVED** [review]
Magnificent! Thanks for all your hard work on this! 🙌 

Noting some things I observed here to keep them in this context. They're not faults with this PR, just points we might want to think about for the future.

### Step 6

I already had WooCommerce Bookings in my cart. When I clicked the subscribe button after trying to update WooCommerce Dynamic Pricing And Discounts, it took me to the cart, but the cart still showed WooCommerce Bookings, not WooCommerce Dynamic Pricing And Discounts. After I removed WooCommerce Bookings and repeated, it worked as expected. I'm sure it's not easy to clear the cart in this situation.

### Step 7

On going back directly to My Subscriptions after purchasing, the list isn't updated, as the user might expect. We can't do much about people viewing the URL directly, but maybe we can think about passing a param when you go back by clicking on a link.

image

After clicking the "update" link to refresh the My Subscriptions list, the "Update" button for the extension is still visible, because all you've done is subscribe. But you're really trying to update. Ideally, after the user has subscribed, we would either update the plugin automatically, or prompt them to update at this stage.

image

**Comment by bgrgicak** [comment]
>  still showed WooCommerce Bookings, not WooCommerce Dynamic Pricing And Discounts. 

That's odd. This code is using standard Woo `add-to-cart`, so it can't do anything custom.
We could replace this with a custom solution in the future.

> On going back directly to My Subscriptions after purchasing, the list isn't updated, as the user might expect

We can't do any of this today. What I would suggest that in the future Woo.com returns a query string to trigger a product install/update.
This would reload the subscription list, and open a popup that will allow the user to click install/update.

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/subscribe-button.tsx (hunk 2 above)
Comment (bgrgicak) [reply plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/subscribe-button.tsx]: I left this by mistake, it's updated now.

`~` is the project home, in our case `plugins/woocommerce-admin/client`.

**Pull request merged**